source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2386,6 +2386,48 @@ lldb::addr_t SwiftLanguageRuntime::FixupAddress(lldb::addr_t addr,
   return addr;
 }
 
+const swift::reflection::TypeInfo *
+SwiftLanguageRuntime::GetTypeInfo(CompilerType type) {
+  swift::CanType swift_can_type(GetCanonicalSwiftType(type));
+  ConstString mangled_name(type.GetMangledTypeName());
+  StringRef mangled_no_prefix =
+      swift::Demangle::dropSwiftManglingPrefix(mangled_name.GetStringRef());
+  swift::Demangle::Demangler Dem;
+  auto demangled = Dem.demangleType(mangled_no_prefix);
+  auto *type_ref = swift::Demangle::decodeMangledType(
+      reflection_ctx->getBuilder(), demangled);
+  if (!type_ref)
+    return nullptr;
+  return reflection_ctx->getBuilder().getTypeConverter().getTypeInfo(type_ref);
+}
+
+bool SwiftLanguageRuntime::IsIndirectResilientType(CompilerType type) {
+  swift::CanType swift_can_type(GetCanonicalSwiftType(type));
+  swift::NominalType *nominal_type =
```

Comment:
Actually you don't need to reject non-nominal types or non-resilient nominals here. Consider these cases:

- A tuple containing two resilient structs
- A Fixed-layout struct containing a resilient struct

In both of these cases the decision to use the inline buffer storage or external storage still depends only on the size of the type.

I think it will all work out because you know statically if you have an inline buffer or not. In the case where you don't have a buffer there's no indirection (regardless of the resilience attribute on the type, etc). If there's a buffer, only the size determines whether there is indirection.

Reply:
> Actually you don't need to reject non-nominal types or non-resilient nominals here. 

isResilient() is a method of NominalTypeDecl and AbstractStorageDecl. How would I even query it for a tuple?

I was banking on fixed-size struct members already being handled by GetMemberVariableOffset but it's probably best to at least test this.